src/main/java/com/squareup/javapoet/Util.java

```diff
@@ -109,4 +110,19 @@ public static void requireExactlyOneOf(Set<Modifier> modifiers, Modifier... mutu
   public static boolean hasDefaultModifier(Collection<Modifier> modifiers) {
     return DEFAULT != null && modifiers.contains(DEFAULT);
   }
+
+  public static String identifier(String part) {
+    checkArgument(part != null, "part == null");
+    checkArgument(!part.isEmpty(), "part is empty");
+    checkArgument(Character.isJavaIdentifierStart(part.charAt(0)), "part[0] is not valid");
+    StringBuilder builder = new StringBuilder();
+    for (int i = 0; i < part.length(); i++) {
+      builder.append(part.charAt(i));
+      if (!SourceVersion.isIdentifier(builder.toString())) {
```

Comment:
Probably not necessary to keep a string builder, you can just do successive calls to `part.substring(0, i)`. With the builder you make more work for yourself by having to pop.